nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/ChildFieldPath.java

```diff
@@ -41,19 +42,43 @@ private FieldValue missingChild(final FieldValue parent) {
         return new StandardFieldValue(null, field, parent);
     }
 
+    private FieldValue missingChild(final FieldValue parent, final RecordField childField) {
+        final RecordField field = new RecordField(childName, childField.getDataType(), childField.isNullable());
+        return new StandardFieldValue(null, field, parent);
+    }
+
     private FieldValue getChild(final FieldValue fieldValue) {
         if (!Filters.isRecord(fieldValue)) {
             return missingChild(fieldValue);
         }
 
         final Record record = (Record) fieldValue.getValue();
         if(record == null) {
-            return missingChild(fieldValue);
+            RecordField parent = fieldValue.getField();
+            if (parent != null) {
+                if (parent.getDataType() instanceof RecordDataType) {
+                    Optional<RecordField> childFieldOptional = ((RecordDataType) parent.getDataType()).getChildSchema().getField(childName);
+                    if (childFieldOptional.isPresent()) {
```

Comment:
If you like functional style - I like that it's more compact - this can be replaced by:

`childFieldOptional.map(recordField -> missingChild(fieldValue, recordField)).orElseGet(() -> missingChild(fieldValue));`